src/commands/builds/ls.ts

```diff
@@ -0,0 +1,179 @@
+import { Flags } from '@oclif/core';
+import chalk from 'chalk';
+import Table from 'cli-table';
+
+import { ApifyCommand } from '../../lib/apify_command.js';
+import { prettyPrintStatus } from '../../lib/commands/pretty-print-status.js';
+import { resolveActorContext } from '../../lib/commands/resolve-actor-context.js';
+import { error, simpleLog } from '../../lib/outputs.js';
+import { getLoggedClientOrThrow, objectGroupBy, ShortDurationFormatter } from '../../lib/utils.js';
+
+const tableFactory = (compact = false) => {
+	const options: Record<string, unknown> = {
+		head: ['Number', 'ID', 'Status', 'Took'],
+		style: {
+			head: ['cyan', 'cyan', 'cyan', 'cyan'],
+			compact,
+		},
+	};
+
+	if (compact) {
+		options.chars = {
+			'mid': '',
+			'left-mid': '',
+			'mid-mid': '',
+			'right-mid': '',
+			middle: ' ',
+			'top-mid': '─',
+			'bottom-mid': '─',
+		};
+	}
+
+	return new Table<[string, string, string, string]>(options);
+};
+
+export class BuildLsCommand extends ApifyCommand<typeof BuildLsCommand> {
+	static override description = 'Lists all builds of the actor.';
+
+	static override flags = {
+		actor: Flags.string({
+			description:
+				'Optional Actor ID or Name to list builds for. By default, it will use the Actor from the current directory',
+		}),
+		offset: Flags.integer({
+			description: 'Number of builds that will be skipped.',
+			default: 0,
+		}),
+		limit: Flags.integer({
+			description: 'Number of builds that will be listed.',
+			default: 10,
+		}),
+		desc: Flags.boolean({
+			description: 'Sort builds in descending order.',
+			default: false,
+		}),
+		compact: Flags.boolean({
+			description: 'Display a compact table.',
+			default: false,
+			char: 'c',
+		}),
+	};
+
+	static override enableJsonFlag = true;
+
+	async run() {
```

Comment:
Please split the method into smaller chunks.

Reply:
This request is too vague. Why? (And don't look at call or run 😓, those predate me but I have been wanting to rewrite and shorten them for a while now)

What even is there to split into smaller chunks? Would love some callouts so I can understand better

Reply:
That is a tough question to answer for someone who didn't write this :smile: 

My reasoning is that there is a lot of different variables in use (~20?), and the human brain can only hold like five "concepts" at once. If you have small functions where it's clear what goes in and what goes out, you can restructure the `run` method so that you can tell at a glance what the steps are and what's there output, and you can also zoom in into each step and check all the nasty details.

Reply:
I tried some splitting up, but unfortunately I don't know how else to split it up more than just how it is right now 😅

Reply:
Well, such is life. No need to stall the PR then, but could we make an issue to track this and possible all the other instances of loooong `run` methods?

Reply:
Sure, but its probably gonna end up in the icebox of the icebox 😅